backend/location.py

```diff
@@ -40,15 +41,62 @@ def leap_eligible(board, hex1, hex2):
     u = (1/number_of_tiles) * displacement
     return not (None in [find_hex(board, hex2.location + i*u) for i in range(number_of_tiles)])
 
-def linked_hexes(board, starting_hex)
-    #given a hex, return the list of all hexes which can be connected to the starting hex by a path of monochromatic auras
-    
+def linked_search(board, starting_hex, check_auras, check_boundary):
+    # given a hex, return the list of all hexes connected to the starting hex
+    # if check_auras, return the list of hexes connected to the starting hex by monochromatic auras
+    # breadth-first search
+    visited_hexes = [starting_hex]
+    # a list of lists of hexes found at each step
+    list_of_steps = [[starting_hex]]
+    new_hex_list = []
+    boundary = []
+    while list_of_steps[-1] != []:
+        for current_hex in list_of_steps[-1]:
+            #iterate over hexes found in the last step
+            for candidate_hex in find_adjacent_hexes(board, current_hex):
+                #check if the current hex's neighbors are the same aura as the start
```

Comment:
this would be more accurate as
`#check if the current hex's neighbors have not yet been visited`